core/src/doc/lives.rs

```diff
@@ -32,146 +32,169 @@ impl Document {
+			lqctx.add_value("access", sess.pick(AC.as_ref()).into());
+			lqctx.add_value("auth", sess.pick(RD.as_ref()).into());
+			lqctx.add_value("token", sess.pick(TK.as_ref()).into());
+			lqctx.add_value("session", sess.clone().into());
+			// Add $before, $after, $value, and $event params
+			// to this LIVE query so the user can use these
+			// within field projections and WHERE clauses.
+			lqctx.add_value("event", met.into());
+			lqctx.add_value("value", current.clone());
+			lqctx.add_value("after", current);
+			lqctx.add_value("before", initial);
+			// We need to create a new options which we will
+			// use for processing this LIVE query statement.
+			// This ensures that we are using the auth data
+			// of the user who created the LIVE query.
+			let lqopt = opt.new_with_perms(true).with_auth(Arc::from(auth));
+			// First of all, let's check to see if the WHERE
+			// clause of the LIVE query is matched by this
+			// document. If it is then we can continue.
+			let lqctx = lqctx.freeze();
+			match self.lq_check(stk, &lqctx, &lqopt, &lq, doc).await {
+				Err(Error::Ignore) => continue,
+				Err(e) => return Err(e),
+				Ok(_) => (),
+			}
+			// Secondly, let's check to see if any PERMISSIONS
+			// clause for this table allows this document to
+			// be viewed by the user who created this LIVE
+			// query. If it does, then we can continue.
+			match self.lq_allow(stk, &lqctx, &lqopt, &lq, doc).await {
+				Err(Error::Ignore) => continue,
+				Err(e) => return Err(e),
+				Ok(_) => (),
+			}
+			// Finally, let's check what type of statement
+			// caused this LIVE query to run, and send the
+			// relevant notification based on the statement.
+			if stm.is_delete() {
+				// Send a DELETE notification
+				if opt.id()? == lv.node.0 {
+					// Ensure futures are run
+					let lqopt: &Options = &lqopt.new_with_futures(true);
+					// Output the full document before any changes were applied
+					let mut result =
+						doc.doc.as_ref().compute(stk, &lqctx, lqopt, Some(doc)).await?;
+					// Remove metadata fields on output
+					result.del(stk, &lqctx, lqopt, &*META).await?;
+					let res = chn
+						.send(Notification {
 							id: lv.id,
 							action: Action::Delete,
 							record: Value::Thing(rid.as_ref().clone()),
-							result: {
-								// Ensure futures are run
-								let lqopt: &Options = &lqopt.new_with_futures(true);
-								// Output the full document before any changes were applied
-								let mut value =
-									doc.doc.as_ref().compute(stk, &lqctx, lqopt, Some(doc)).await?;
-								// Remove metadata fields on output
-								value.del(stk, &lqctx, lqopt, &*META).await?;
-								// Output result
-								value
-							},
+							result,
 						})
-						.await?;
-					} else {
-						// TODO: Send to message broker
+						.await;
+
+					if res.is_err() {
+						// channel was closed, that means a transaction probably failed.
+						// just return as nothing can be send.
+						return Ok(());
 					}
```

Comment:
This check was added to ensure that we don't error when a channel is closed.

Previously notification channels would be reused between transactions but that might lead to notification of canceled transaction ending up between correct notifications.